.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,110 @@
+name: Unit
+
+on:
+  push:
+  pull_request:
+
+jobs:
+
+  test:
+    name: Tests
+    runs-on: Ubuntu-20.04
+
+    env:
+      extensions: amqp,apcu,igbinary,intl,mbstring,memcached,mongodb,redis
+      EXCLUDE_GROUPS: benchmark,intl-data,tty,integration
+
+    strategy:
+      matrix:
+        include:
+            - php: '7.1'
+            - php: '7.2'
+            - php: '7.3'
+            - php: '7.4'
+              deps: high
+            - php: '8.0'
+              deps: low
+            - php: '8.1'
+              deps: experimental
+      fail-fast: false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure for PHP 8.1
+        if: "${{ matrix.php == '8.1' }}"
+        run: |
+          composer config platform.php 8.0.99
+          echo "extensions=mbstring" >> $GITHUB_ENV
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: "none"
+          ini-values: date.timezone=Europe/Paris,memory_limit=-1,default_socket_timeout=10,session.gc_probability=0,apc.enable_cli=1
+          php-version: "${{ matrix.php }}"
+          extensions: "${{ env.extensions }}"
+          tools: flex
+
+      - name: Install dependencies
+        run: |
+          COMPOSER_HOME="$(composer config home)"
+          ([ -d "$COMPOSER_HOME" ] || mkdir "$COMPOSER_HOME") && cp .github/composer-config.json "$COMPOSER_HOME/config.json"
+          export COMPOSER_ROOT_VERSION=$(grep ' VERSION = ' src/Symfony/Component/HttpKernel/Kernel.php | grep -P -o '[0-9]+\.[0-9]+').x-dev
+          export SYMFONY_REQUIRE=">=$COMPOSER_ROOT_VERSION"
+          echo "COMPOSER_ROOT_VERSION=$COMPOSER_ROOT_VERSION" >> $GITHUB_ENV
+          echo "SYMFONY_REQUIRE=$SYMFONY_REQUIRE" >> $GITHUB_ENV
+
+          echo "::group::composer update"
+            composer require --dev --no-update masterminds/html5:~2.7.5@dev
+            composer update --no-progress --ansi
+          echo "::endgroup::"
+
+          echo "::group::install phpunit"
+            ./phpunit install
+          echo "::endgroup::"
+
+      - name: Patch return types
+        if: "${{ matrix.php == '8.0' }}"
```

Comment:
shouldn't we also patch them on PHP 8.1 ?

Reply:
we just need one job to check return types